### PR TITLE
Disable auth in iframe scenarios (e.g. skillmap)

### PR DESCRIPTION
**Pull request by darzu**
Fixes: https://github.com/microsoft/pxt-arcade/issues/3150

### CODE REVIEW

File: webapp/src/app.tsx [hunk 1]
```diff
@@ -4698,11 +4698,11 @@ document.addEventListener("DOMContentLoaded", () => {
         auth.loginCallback(query);
     }
 
+    auth.init();
     // Disable auth in skillmap until it is properly supported.
     // See https://github.com/microsoft/pxt-arcade/issues/3138
     const disableAuth = query["skillsMap"] == "1";
```
[the remark this answers is not in the document]
Comment (darzu) [reply webapp/src/app.tsx:4704]: Actually why wasn't this working? When was "skillsMap" there?

File: webapp/src/app.tsx [hunk 2]
```diff
@@ -4741,6 +4741,9 @@ document.addEventListener("DOMContentLoaded", () => {
     else if (pxt.BrowserUtils.isIpcRenderer()) workspace.setupWorkspace("idb");
     else if (pxt.BrowserUtils.isPxtElectron()) workspace.setupWorkspace("fs");
     else workspace.setupWorkspace("browser");
+    // disable auth in iframe scenarios
+    if (workspace.getWorkspaceType() === "iframe")
```
Comment (eanders-MS) [inline webapp/src/app.tsx:4745]: In the future we might want to allow auth but disable cloud sync for the iframe workspace.

Comment (darzu) [reply webapp/src/app.tsx:4745]: yeah.

File: webapp/src/app.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (eanders-MS) [reply webapp/src/app.tsx]: It worked when I tested it! Maybe it was removed? This is a brittle check anyway with no support from the compiler. Guaranteed to break eventually.

**Comment by darzu** [comment]
This is hard to test on localhost I think.
I am running `pxt serve` and `pxt skillmap --serve` and I can successfully open a skillmap tutorial make changes and see that those aren't being synced to the main homescreen. I think that confirms it?

**Comment by darzu** [comment]
> This is hard to test on localhost I think.
> I am running `pxt serve` and `pxt skillmap --serve` and I can successfully open a skillmap tutorial make changes and see that those aren't being synced to the main homescreen. I think that confirms it?

nvm, it's not repro'ing on master using this technique. I don't know how to confirm this fixes skillsmap.
I think the problem is that normal pxt is running on localhost:3232 and skillsmap is running at localhost:3000 so they have different dbs.

Any ideas @riknoll , @shakao ?

File: webapp/src/app.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (darzu) [reply webapp/src/app.tsx]: yeah.

I'm having trouble testing this fix on localhost. I'm not able to repro the issue (on master) locally.
Want to try it or explain ur method?

**Comment by eanders-MS** [comment]
> let authDisabled = true;

Start with `authDisabled = false` now, then the app can optionally disabled it. Or maybe remove the double-negative and called it`authEnabled`.

---
Refers to: webapp/src/auth.ts:28 in f3b7cb5. [](commit_id = f3b7cb59ad78f3b22a0de88ea12515657ef929f8, deletion_comment = False)

**Review by eanders-MS — APPROVED** [review]
:shipit:

**Pull request merged**